base_exception/models/base_exception.py

```diff
@@ -63,6 +72,19 @@ class ExceptionRule(models.Model):
 #  - context: current context
 """)
 
+    @api.multi
+    def _get_domain(self):
+        """ override me to customize domains according exceptions cases """
+        self.ensure_one()
+        return safe_eval(self.domain)
```

Comment:
I don't think the domain should be evaled

Reply:
I don't like safe_eval, but it seams to fit with its docstring 

>     This can be used to e.g. evaluate an OpenERP domain expression from an untrusted source.

https://github.com/odoo/odoo/blob/10.0/odoo/tools/safe_eval.py#L267
And there is some use of it across the sources

Reply:
Yes, but this is introducing a great security risk, enabling everyone that access these records to do what they want with the DB. You can write a domain like: `self.sudo().password = 'x'` and gain admin access with the specific password you have changed

Always remember the dear BOBBY TABLES:

![imagen](https://user-images.githubusercontent.com/7165771/49648115-f7ebea00-fa24-11e8-88d4-85f2a894b2a9.png)

Or you find another way of representing safely domains, or for me this is :-1:

Reply:
I agree about how dangerous is to allow user execute code, even in a "safe_eval".
But, 
* I don't see if is more dangerous than allowing user write python code (default version of the module) 
* I don't know a safer way

Reply:
Yes, indeed this is the same hole as with python code. What about this:

* Put a constraint over this field.
* The constraint will add under a savepoint that:

  * There's no `sudo` word inside the text
  * When evaled, it's a valid domain (I'm not sure if there's a method for checking this), but maybe we can perform the search with it to see it.

I know it's limited, but it can be a good first step.

Reply:
It's too much work to fix : `getattr(self, 'odus'[::-1])() `

IMO, the only way to fix it, is to require python or domain rules to be hard coded in class files. It's secure by default. User input rules should be allowed by another optional module. 

It's out of scope of the current PR.

Reply:
OK, at least add this debate in the known issues / roadmap section.